### PR TITLE
Make StrictContextStorage package-private

### LINKED ISSUE #2735: Hide StrictContextStorage implementation, expose only the flag to configure it
If we do that we can remove the extra shaded dependencies from the Context package and the package becomes way smaller and simpler. I think this will be a huge win especially because we expect the Context package to be used the more.

Advantages like the number of classes we export (for platforms like Android) will be very appreciated by the users.

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #2735

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
👍 could you throw an item into the CHANGELOG for this as well?

**Comment by anuraaga** [comment]
Sure sent #2773

File: extensions/kotlin/build.gradle [hunk 1]
```diff
@@ -29,4 +34,8 @@ tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
 }
 
 // We don't have any public Java classes
-tasks.javadoc.enabled = false
\ No newline at end of file
+tasks.javadoc.enabled = false
+
+testStrictContext {
+    jvmArgs("-Dio.opentelemetry.context.enableStrictContext=true")
+}
```
Comment (bogdandrutu) [inline extensions/kotlin/build.gradle:41]: New line?

**Review by bogdandrutu — APPROVED** [review]

File: extensions/kotlin/build.gradle (hunk 1 above)
Comment (bogdandrutu) [reply extensions/kotlin/build.gradle:41]: Can be done in a separate PR :)

**Pull request merged**